examples/model_selection/plot_tuned_decision_threshold.py

```diff
@@ -157,31 +165,192 @@
+    return decision_threshold.mean()
+
+
+mean_threshold = process_decision_thresholds(cv_results_tuned_model)
+mean_threshold_constrained = process_decision_thresholds(
+    cv_results_constrained_metric_model
 )
-ax.set_xlabel("Decision threshold")
-ax.legend(loc="upper right")
-_ = ax.set_title(
-    "Distribution of the decision threshold \nacross different cross-validation folds"
+
+_ = ax.legend(
+    [
+        "Tuned Model",
+        f"Mean Decision Threshold : {mean_threshold:.2f}",
+        "Tuned Model (Constrained Metric)",
+        f"Mean Decision Threshold : {mean_threshold_constrained:.2f}",
+    ]
 )
 
 # %%
-# In average, a decision threshold around 0.32 maximizes the balanced accuracy, which is
-# different from the default decision threshold of 0.5. Thus tuning the decision
-# threshold is particularly important when the output of the predictive model
-# is used to make decisions. Besides, the metric used to tune the decision threshold
-# should be chosen carefully. Here, we used the balanced accuracy but it might not be
-# the most appropriate metric for the problem at hand. The choice of the "right" metric
-# is usually problem-dependent and might require some domain knowledge. Refer to the
-# example entitled,
+# On average, a decision threshold around 0.32 maximizes the balanced accuracy,
+# and a decision threshold of 0.43 maximizes the constrained balanced accuracy.
+# These are different from the default decision threshold of 0.5. Thus tuning the
+# decision threshold is particularly important when the output of the predictive
+# model is used to make decisions. Besides, the metric used to tune the decision
+# threshold should be chosen carefully. Here, we used the balanced accuracy but it
+# might not be the most appropriate metric for the problem at hand. The choice of the
+# "right" metric is usually problem-dependent and might require some domain knowledge.
+# Refer to the example entitled,
 # :ref:`sphx_glr_auto_examples_model_selection_plot_cost_sensitive_learning.py`,
 # for more details.
+
+
+# We conclude this section by plotting the mean ROC curve and the points on the curve
+# corresponding to our classifiers. To do the latter, we will compute the FPR and
+# TPR on each test set passed to our classifiers. TPR is the same as recall,
+# which we already have. FPR can be obtained from the TPR and balanced accuracy,
+# recalling that `:math:`FPR=1 - TNR` and
+# :math:`\text{Balanced Accuracy}\,=\frac{TPR+TNR}{2}`
+def compute_point_roc(cv_model_results):
+    ba = cv_model_results["test_balanced_accuracy"].mean()
+    tpr = cv_model_results["test_recall"].mean()
+    return (1 - (2 * ba - tpr), tpr)
+
+
+roc_points = {
+    "vanilla": compute_point_roc(cv_results_vanilla_model),
+    "max_balanced_accuracy": compute_point_roc(cv_results_tuned_model),
+    "constrained_balanced_accuracy": compute_point_roc(
+        cv_results_constrained_metric_model
+    ),
+}
+# %%
```

Comment:
While the plot is cool because it is using CV (we try to get a `.from_cv_results`) I think that we need to drop it.
Instead, we are better the same type of analysis with the precision recall curve since we are going to compute the score for the precision recall and f1-score.